src/app/common/services/alert.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from '@angular/core';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+export enum AlertType {
+  INFO,
+  SUCCESS,
+  WARNING,
+  DANGER,
+}
+@Injectable({
+  providedIn: 'root',
+})
+export class AlertService {
+  constructor(private snackBar: MatSnackBar) {}
+  add(type: AlertType, message: string, duration?: number, action = 'Dismiss') {
+    switch (type) {
+      case AlertType.INFO:
+        this.snackBar.open(message, action, {
+          duration: duration ? duration : 3000,
+          horizontalPosition: 'end',
+          verticalPosition: 'top',
+          panelClass: ['alert-info'],
+        });
+        break;
+      case AlertType.SUCCESS:
+        this.snackBar.open(message, action, {
+          duration: duration ? duration : 5000,
+          horizontalPosition: 'end',
+          verticalPosition: 'top',
+          panelClass: ['alert-success'],
+        });
+        break;
+      case AlertType.WARNING:
+        this.snackBar.open(message, action, {
+          duration: duration ? duration : 6000,
+          horizontalPosition: 'end',
+          verticalPosition: 'top',
+          panelClass: ['alert-warning'],
+        });
+        break;
+      case AlertType.DANGER:
+        this.snackBar.open(message, action, {
+          duration: duration ? duration : 8000,
+          horizontalPosition: 'end',
+          verticalPosition: 'top',
+          panelClass: ['alert-danger'],
+        });
+        break;
+    }
+  }
+}
```

Comment:
Can we refactor this to have 4 methods called danger, warning, success and info. This means we no longer need the AlertType type to be known or used in the various callers of the AlertService.

We also only need one private function which opens the snackbar

Reply:
commit: https://github.com/doubtfire-lms/doubtfire-web/pull/650/commits/62ac3e891f326d92b52d46bbc9b6208a2e96dbec